### DISCUSSION TITLE
`parquet_file_metadata` function is not available in duckdb?

### DISCUSSION THREAD

**Discussion by llimllib** [observablehq/framework#1813] [Q&A] [answered]
In the [duckdb shell](https://shell.duckdb.org/), which I believe is running the same version of duckdb wasm as observable framework, the `parquet_file_metadata` function is available:
```
duckdb> select created_by from parquet_file_metadata('https://raw.githubusercontent.com/llimllib/nba_data/refs/heads/main/data/playerstats.parquet');
┌──────────────────────────────────┐
│ created_by                       │
╞══════════════════════════════════╡
│ parquet-cpp-arrow version 13.0.0 │
└──────────────────────────────────┘
```

However, in observable framework document, that same command fails with:

```
Error: Catalog Error: Table Function with name parquet_file_metadata does not exist!
Did you mean "parquet_metadata"?
LINE 1: select created_by from parquet_file_metadata('https://raw.gith...
```

Does anybody know why this function seems to be missing from observable, but not from the official wasm build?

cf [duckdb docs](https://duckdb.org/docs/data/parquet/metadata.html), [this issue](https://github.com/duckdb/duckdb/issues/2534), resolved [here](https://github.com/duckdb/duckdb/pull/9793) and [here](https://github.com/duckdb/duckdb/pull/9798), released in [0.10.0](https://github.com/duckdb/duckdb/releases/tag/v0.10.0), which should be in the version used in framework (which [corresponds to release 1.1.1](https://observablehq.com/framework/lib/duckdb#versioning))

**Comment by mbostock** [comment, marked as answer]
Are you using the prerelease version of Framework by building from the main branch? We haven’t yet released support for DuckDB 1.1.1/DuckDB-Wasm 1.29.0.

**Reply by llimllib** [reply]
ah ha! no I'm on `1.12.0`, whose docs say nothing about the version of duckdb in use; I didn't know that the docs site was for pre-release versions.

Do I need to upgrade to HEAD if I want to use an updated duckdb?

**Reply by Fil** [reply]
It is possible (I published a how-to [here](https://github.com/observablehq/framework/discussions/1768#discussioncomment-11084080)) but I'd suggest to wait a little bit as the new release is almost ready.

PS: I tried your query it returns `parquet-cpp-arrow version 18.0.0-SNAPSHOT`.

**Reply by llimllib** [reply]
very cool, thanks for the answer!
